### DISCUSSION TITLE
为什么两个Select进行Join时会报错

### DISCUSSION THREAD

**Discussion by AuroraRyan** [physics-data/faq#153] [Q&A] [answered]
`uni = ''' 
SELECT year, athlete, gender, event, medal FROM events
LEFT JOIN
SELECT * FROM athletes
ON events.athlete=athletes.athlete AND events.gender=athletes.gender AND events.year=athletes.year
'''`
报错信息

> OperationalError                          Traceback (most recent call last)
<ipython-input-133-9b195a6483b5> in <module>
----> 1 e=c.execute(uni)

> OperationalError: near "SELECT": syntax error

网上找到了两个Select进行Join的例子，是与sql版本有关吗？

此外去除第二个表的select后，即改成

`uni = '''
SELECT year, athlete, gender, event, medal FROM events
LEFT JOIN
athletes
ON events.athlete=athletes.athlete AND events.gender=athletes.gender AND events.year=athletes.year
'''`

后，报错变为：

> OperationalError                          Traceback (most recent call last)
<ipython-input-131-9b195a6483b5> in <module>
----> 1 e=c.execute(uni)

> OperationalError: ambiguous column name: year

但是，如果Join 前，再改成全选"select * from events“，则不会报错，不太能理解其中的原因

**Comment by Harry-Chen** [comment, marked as answer]
> `OperationalError: near "SELECT": syntax error`

这意味着语法错误，你搜到的不是 SQLite 支持的语法。并且这样写是不高效的。

> `OperationalError: ambiguous column name: year` 

这是因为两个表里面都有 `year` 列，你需要指定是哪里的 `year`。关于你说的如果改成 `SELECT *` 就不会报错，我没有仔细调研。这可能是 SQLite 的某种特性，但是这样书写 SQL 是不推荐的，容易造成问题。

**Reply by AuroraRyan** [reply]
懂了，谢谢助教
